nav2_dwb_controller/nav_2d_utils/include/nav_2d_utils/odom_subscriber.hpp

```diff
@@ -77,14 +81,26 @@ class OdomSubscriber
     // ROS_INFO_ONCE("odom received!");
     std::lock_guard<std::mutex> lock(odom_mutex_);
     odom_vel_.header = msg->header;
-    odom_vel_.velocity.x = msg->twist.twist.linear.x;
-    odom_vel_.velocity.y = msg->twist.twist.linear.y;
-    odom_vel_.velocity.theta = msg->twist.twist.angular.z;
+    odom_vel_.velocity.x =
+      thresholded_velocity(msg->twist.twist.linear.x, min_x_velocity_threshold_);
```

Comment:
What happens if any of these thresholds are negative numbers? I see the thresholded_velocity seems to assume they aren't but there's no guarantee someone hasn't put a -1.0 in the param file. Should the absolute value of the threshold be used just like the absolute value of the velocity?

Reply:
> What happens if any of these thresholds are negative numbers?

If any threshold is negative, the result would be as if they had specified 0. We check if the absolute value of the velocity is greater than the threshold. For 0 or any negative value, this condition would always be true.

I think this is reasonable behavior for now. We need to revisit all our parameters and put bounds checking on them soon, so when that is done, we can ensure the threshold values are positive.